File: frontend/src/features/auth/http/request/RegisterRequest.ts
import type { DisplayName, Username } from '@/features/user/model/UserValues'
import type { EmailAddress, PlaintextPassword } from '@/features/auth/model/AuthValues'

export type RegisterRequest = {
  username: Username
  displayName: DisplayName
  email: EmailAddress
  password: PlaintextPassword
}

File: backend/src/main/scala/domains/auth/http/request/RegisterRequest.scala
package domains.auth.http.request

import domains.auth.model.*

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class RegisterRequest(
  username: Username,
  displayName: DisplayName,
  email: EmailAddress,
  password: PlaintextPassword
)

object RegisterRequest:
  given Encoder[RegisterRequest] = deriveEncoder[RegisterRequest]
  given Decoder[RegisterRequest] = deriveDecoder[RegisterRequest]
